src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -418,28 +448,157 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
     else:
         raise CLIError('DNS routing verification failed')
 
+# Validate --image input
+def validate_image(cmd,
+                   registry_name,
+                   repository_name,
+                   login_server,
+                   username,
+                   password,
+                   image):
+    from .manifest import _get_v2_manifest_path
+
+    from ._docker_utils import (
+        request_data_from_registry,
+        get_manifest_authorization_header,
+        RegistryException
+    )
+
+    try:
+        request_data_from_registry(
+            http_method='get',
+            login_server=login_server,
+            path=_get_v2_manifest_path(repository_name, image),
+            username=username,
+            password=password,
+            json_payload=None,
+            file_payload=None,
+            manifest_headers=get_manifest_authorization_header(username, password),
+            params=image)
+
+    except RegistryException as e:
+        if e.status_code == 404:
+            _check_image_match(cmd,
+                               registry_name,
+                               repository_name,
+                               login_server,
+                               username,
+                               password,
+                               image)
+        raise CLIError("{} Please check if image/tag was inputted correctly.".format(str(e)))
+
+# Validate if blob layer can be pulled
```

Comment:
what do you mean by "can be pulled"?

Reply:
@rosanch ' can be pulled' == 'can be downloaded'

Reply:
My bad, when can it not be pulled it's what I meant.

Reply:
If blob layer can't be downloaded, it could be for a variety of reasons (ie. requested data not found, unauthorized, insufficient permissions etc.) and if that's the case, the server error message should return a status code and message that specifies the reason.